base/actions/actions.py

```diff
@@ -704,14 +706,171 @@ async def submit(
                     if i == config.HTTP_RETRIES - 1:
                         raise e
 
-        for template in templates:
+        if group_arm:
+            if group_arm.startswith("soft_"):
+                return [SlotSet("group_arm", group_arm)]
             dispatcher.utter_message(template=template)
+        return []
+
+
+class GroupArmForm(BaseFormAction):
+    YES_NO_MAPPING = {
+        "yes": True,
+        "no": False,
+    }
+
+    GENDER_MAPPING = {
```

Comment:
why do you need to define all these class properties in this class too?

Reply:
I use them to map user response options, as they are required fields on api

Reply:
No longer needed as i do patch request